api/action/index.js

```diff
@@ -0,0 +1,233 @@
+/**
+ *
+ * @param {number} timeout
+ * @param {function} callback
+ * @returns
+ */
+function debounce(timeout, callback) {
+  let timeoutID;
+
+  return function(event) {
+    if (timeoutID) {
+      clearTimeout(timeoutID);
+    }
+
+    timeoutID = setTimeout(() => {
+      timeoutID = 0;
+      callback(event);
+    }, timeout);
+  }
+}
+
+// ------------------
+// .enable / .disable
+// ------------------
+
+// The action API does not expose a way to read the action's current enabled/disabled state, so we
+// have to track it ourselves.
+// Relevant feature request: https://bugs.chromium.org/p/chromium/issues/detail?id=1189295
+let actionEnabled = true;
+let showToggleState = document.getElementById('show-toggle-state');
+document.getElementById('toggle-state-button').addEventListener('click', (_event) => {
+  if (actionEnabled) {
+    chrome.action.disable();
+  } else {
+    chrome.action.enable();
+  }
+  actionEnabled = !actionEnabled;
+});
+
+document.getElementById('popup-options').addEventListener('change', async (event) => {
+  let popup = event.target.value;
+  await chrome.action.setPopup({ popup });
+
+  // Show the updated popup path
+  await getCurrentPopup();
+});
+
+async function getCurrentPopup() {
+  let popup = await chrome.action.getPopup({});
+  document.getElementById('current-popup-value').value = popup;
+  return popup;
+};
+
+async function showCurrentPage() {
+  let popup = await getCurrentPopup();
+  if (!popup) return;
+  let url = new URL(popup);
+
+  let options = document.getElementById('popup-options');
+  let option = options.querySelector(`option[value="${url.pathname}"]`);
+  option.selected = true;
+}
+
+// Populate popup inputs on on page load
+showCurrentPage();
+
+// ----------
+// .onClicked
+// ----------
+
+// If a popup is specified, our on click handler won't be called. We declare it here rather than in
+// the `onclicked-button` handler to prevent the user from accidentally registering multiple
+// onClicked listeners.
+chrome.action.onClicked.addListener((tab) => {
+  chrome.tabs.create({ url: 'https://html5zombo.com/' });
+});
+
+document.getElementById('onclicked-button').addEventListener('click', async () => {
+  // Our listener will only receive the action's click event after clear out the popup URL
+  await chrome.action.setPopup({ popup: '' });
+  await showCurrentPage();
```

Comment:
My minor complaint here is that this action makes the `<select>` not really show the correct state any more (since there's default, A, B.). But it's so minor I'm not even that worried.

Reply:
There's already other stuff to clean up, so I'll integrate this now.